lib/ex_limiter/plug.ex

```diff
@@ -97,17 +97,25 @@ defmodule ExLimiter.Plug do
         remaining = @limiter.remaining(bucket, scale: scale, limit: limit)
 
         conn
-        |> put_resp_header("x-ratelimit-limit", to_string(limit))
-        |> put_resp_header("x-ratelimit-window", to_string(scale))
-        |> put_resp_header("x-ratelimit-remaining", to_string(remaining))
+        |> put_rate_limit_headers(limit, scale, remaining)
         |> decorate_fun.(response)
 
       {:error, :rate_limited} ->
+        remaining = @limiter.remaining(%ExLimiter.Bucket{key: bucket_name}, scale: scale, limit: limit)
+
         conn
+        |> put_rate_limit_headers(limit, scale, remaining)
         |> decorate_fun.({:rate_limited, bucket_name})
         |> fallback.render_error(:rate_limited)
     end
   end
 
+  defp put_rate_limit_headers(conn, limit, scale, remaining) do
+    conn
+    |> put_resp_header("x-ratelimit-limit", to_string(limit))
+    |> put_resp_header("x-ratelimit-window", to_string(scale))
+    |> put_resp_header("x-ratelimit-remaining", to_string(remaining))
```

Comment:
if you know these are ints (??) you could get a minor optimization + reader legibility increase by using `Integer.to_string`

Reply:
Every indication is that these figures are integers, so that works. 

Updated in 0c5cd1b6131362c1ffe8c17e6e370cad9029fd24

Reply:
cool, my only fear would be if a wild `nil` could appear

Reply:
the `round/1` call in ExLimiter.Base implies that we'd raise way sooner and it will always return an int for `remaining`. Default values configured for `scale` and `limit` are integers. I can't stop library users from putting in goofball values though 🤷